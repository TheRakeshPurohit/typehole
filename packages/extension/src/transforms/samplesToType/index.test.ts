import { samplesToType } from ".";

test("generates types and interfaces from samples", () => {
  expect(samplesToType([1, { a: 2 }, true, null])).toBe(
    `type TypeholeRoot = (boolean | TypeholeRootWrapper2 | null | number);
interface TypeholeRootWrapper2 {
  a: number;
}`
  );
  expect(samplesToType([{ a: 2 }])).toBe(
    `interface TypeholeRoot {
  a: number;
}`
  );
  expect(samplesToType([{ a: 2 }, { a: null }])).toBe(
    `interface TypeholeRoot {
  a?: number;
}`
  );

  expect(samplesToType([require("./redditResponse.json")])).toMatchSnapshot();
});
