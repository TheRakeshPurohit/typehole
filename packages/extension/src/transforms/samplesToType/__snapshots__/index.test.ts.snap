// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates types and interfaces from samples 1`] = `
"interface TypeholeRoot {
  kind: string;
  data: Data2;
}
interface Data2 {
  modhash: string;
  dist: number;
  children: Child[];
  after: string;
  before: null;
}
interface Child {
  kind: string;
  data: Data;
}
interface Data {
  approved_at_utc: null;
  subreddit: string;
  selftext: string;
  author_fullname: string;
  saved: boolean;
  mod_reason_title: null;
  gilded: number;
  clicked: boolean;
  title: string;
  link_flair_richtext: any[];
  subreddit_name_prefixed: string;
  hidden: boolean;
  pwls: number;
  link_flair_css_class: null | string;
  downs: number;
  thumbnail_height: number;
  top_awarded_type: null;
  hide_score: boolean;
  name: string;
  quarantine: boolean;
  link_flair_text_color: string;
  upvote_ratio: number;
  author_flair_background_color: null;
  subreddit_type: string;
  ups: number;
  total_awards_received: number;
  media_embed: Mediaembed;
  thumbnail_width: number;
  author_flair_template_id: null;
  is_original_content: boolean;
  user_reports: any[];
  secure_media: Securemedia;
  is_reddit_media_domain: boolean;
  is_meta: boolean;
  category: null;
  secure_media_embed: Securemediaembed;
  link_flair_text: null | string;
  can_mod_post: boolean;
  score: number;
  approved_by: null;
  author_premium: boolean;
  thumbnail: string;
  edited: boolean;
  author_flair_css_class: null;
  author_flair_richtext: any[];
  gildings: Gildings;
  post_hint: string;
  content_categories: null;
  is_self: boolean;
  mod_note: null;
  created: number;
  link_flair_type: string;
  wls: number;
  removed_by_category: null;
  banned_by: null;
  author_flair_type: string;
  domain: string;
  allow_live_comments: boolean;
  selftext_html: null;
  likes: null;
  suggested_sort: null;
  banned_at_utc: null;
  url_overridden_by_dest: string;
  view_count: null;
  archived: boolean;
  no_follow: boolean;
  is_crosspostable: boolean;
  pinned: boolean;
  over_18: boolean;
  preview: Preview;
  all_awardings: (Allawarding | Allawardings2 | Allawardings3)[];
  awarders: any[];
  media_only: boolean;
  can_gild: boolean;
  spoiler: boolean;
  locked: boolean;
  author_flair_text: null;
  treatment_tags: any[];
  visited: boolean;
  removed_by: null;
  num_reports: null;
  distinguished: null;
  subreddit_id: string;
  mod_reason_by: null;
  removal_reason: null;
  link_flair_background_color: string;
  id: string;
  is_robot_indexable: boolean;
  report_reasons: null;
  author: string;
  discussion_type: null;
  num_comments: number;
  send_replies: boolean;
  whitelist_status: string;
  contest_mode: boolean;
  mod_reports: any[];
  author_patreon_flair: boolean;
  author_flair_text_color: null;
  permalink: string;
  parent_whitelist_status: string;
  stickied: boolean;
  url: string;
  subreddit_subscribers: number;
  created_utc: number;
  num_crossposts: number;
  media: Securemedia;
  is_video: boolean;
  link_flair_template_id?: string;
}
interface Allawardings3 {
  giver_coin_reward: null | number;
  subreddit_id: null;
  is_new: boolean;
  days_of_drip_extension: number;
  coin_price: number;
  id: string;
  penny_donate: null | number;
  award_sub_type: string;
  coin_reward: number;
  icon_url: string;
  days_of_premium: number;
  tiers_by_required_awardings: null;
  resized_icons: Source[];
  icon_width: number;
  static_icon_width: number;
  start_date: null;
  is_enabled: boolean;
  awardings_required_to_grant_benefits: null;
  description: string;
  end_date: null;
  subreddit_coin_reward: number;
  count: number;
  static_icon_height: number;
  name: string;
  resized_static_icons: Source[];
  icon_format: null | string;
  icon_height: number;
  penny_price: null | number;
  award_type: string;
  static_icon_url: string;
}
interface Allawardings2 {
  giver_coin_reward: number;
  subreddit_id: null;
  is_new: boolean;
  days_of_drip_extension: number;
  coin_price: number;
  id: string;
  penny_donate: number;
  award_sub_type: string;
  coin_reward: number;
  icon_url: string;
  days_of_premium: number;
  tiers_by_required_awardings: Tiersbyrequiredawardings;
  resized_icons: Source[];
  icon_width: number;
  static_icon_width: number;
  start_date: null;
  is_enabled: boolean;
  awardings_required_to_grant_benefits: number;
  description: string;
  end_date: null;
  subreddit_coin_reward: number;
  count: number;
  static_icon_height: number;
  name: string;
  resized_static_icons: Source[];
  icon_format: string;
  icon_height: number;
  penny_price: number;
  award_type: string;
  static_icon_url: string;
}
interface Tiersbyrequiredawardings {
  '0': _0;
  '5': _0;
  '10': _0;
  '25': _0;
}
interface _0 {
  resized_icons: Source[];
  awardings_required: number;
  static_icon: Staticicon;
  resized_static_icons: Source[];
  icon: Icon;
}
interface Icon {
  url: string;
  width: number;
  format: string;
  height: number;
}
interface Staticicon {
  url: string;
  width: number;
  format: null;
  height: number;
}
interface Allawarding {
  giver_coin_reward: null;
  subreddit_id: null;
  is_new: boolean;
  days_of_drip_extension: number;
  coin_price: number;
  id: string;
  penny_donate: null;
  award_sub_type: string;
  coin_reward: number;
  icon_url: string;
  days_of_premium: number;
  tiers_by_required_awardings: null;
  resized_icons: Source[];
  icon_width: number;
  static_icon_width: number;
  start_date: null;
  is_enabled: boolean;
  awardings_required_to_grant_benefits: null;
  description: string;
  end_date: null;
  subreddit_coin_reward: number;
  count: number;
  static_icon_height: number;
  name: string;
  resized_static_icons: Source[];
  icon_format: null;
  icon_height: number;
  penny_price: null;
  award_type: string;
  static_icon_url: string;
}
interface Preview {
  images: Image[];
  enabled: boolean;
}
interface Image {
  source: Source;
  resolutions: Source[];
  variants: Variants;
  id: string;
}
interface Variants {
  obfuscated?: Obfuscated;
  nsfw?: Obfuscated;
}
interface Obfuscated {
  source: Source;
  resolutions: Source[];
}
interface Source {
  url: string;
  width: number;
  height: number;
}
interface Gildings {
  gid_1?: number;
  gid_2?: number;
}
interface Securemediaembed {
  content: string;
  width: number;
  scrolling: boolean;
  media_domain_url: string;
  height: number;
}
interface Securemedia {
  oembed: Oembed;
  type: string;
}
interface Oembed {
  provider_url: string;
  title: string;
  html: string;
  thumbnail_width: number;
  height: number;
  width: number;
  version: string;
  author_name: string;
  provider_name: string;
  thumbnail_url: string;
  type: string;
  thumbnail_height: number;
  author_url: string;
  description?: string;
}
interface Mediaembed {
  content: string;
  width: number;
  scrolling: boolean;
  height: number;
}"
`;
